### PR TITLE
Split event in index by past and future.

### PR DESCRIPTION
**Pull request by rhnonose**
Also, fix event sorting:
- was only considering `starting_time` and `finishing_time` since the `date` part is separated
- was `desc` instead of `asc`

### CODE REVIEW

File: lib/course_planner/events/events.ex [hunk 1]
```diff
@@ -17,15 +17,30 @@ defmodule CoursePlanner.Events do
 
   def all do
     query = from e in Event,
-    order_by: [desc: e.starting_time, desc: e.finishing_time]
+    order_by: [asc: e.date, asc: e.starting_time, asc: e.finishing_time]
```
Comment (ghatighorias) [inline lib/course_planner/events/events.ex]: why did you change the ordering to ascending ?

File: lib/course_planner/events/events.ex [hunk 2]
```diff
@@ -17,15 +17,30 @@ defmodule CoursePlanner.Events do
...
 
     Repo.all(query)
   end
 
+  def all_splitted(now) do
```
Comment (ghatighorias) [inline lib/course_planner/events/events.ex]: is there any logic to distinguish between all and the events for a specific person ?
I think again like courses in the index page we should not show other events to people who are not added as participants, but rather provide the option via calendar using "my_classes" option

File: test/course_planner/events/events_test.exs [hunk 3]
```diff
@@ -21,6 +21,20 @@ defmodule CoursePlanner.EventsTest do
       assert Events.all_with_users() == [event |> Repo.preload(:users)]
     end
 
+    test "all_splitted/1 returns events splited by time" do
+      %{id: id1} = insert(:event, %{date: ~D[2017-02-02], starting_time: ~T[10:00:00.000000]})
```
Comment (ghatighorias) [inline test/course_planner/events/events_test.exs]: please use Timex for setting time in future or past or your test becomes obsolete after a while
E.g. `Timex.shift(Timex.now(), days: -10))`

File: lib/course_planner/events/events.ex (hunk 2 above)
Comment (rhnonose) [reply lib/course_planner/events/events.ex]: No, it's not. It was created as a default index and I understood that all the events would be visible, but I guess I was wrong.

File: lib/course_planner/events/events.ex (hunk 1 above)
Comment (rhnonose) [reply lib/course_planner/events/events.ex]: I'm not sure, I had the impression that `desc` was wrong.

Comment (ghatighorias) [reply lib/course_planner/events/events.ex]: This way the earliest event ever to happen always stay on top regardless of the current date, so after create some events, the user should always scroll down.

I have a few suggestion, but do tell me what do you think
1 - to have three sections : (past and upcoming events) . now if past events are sorted Descending and upcoming is sorted ascending the one on the top is whether the one that lately happen or the one that is going to happen the earliest  
2 - to sort everything descending, this i have applied in other places and solve the problem for a while
3 - to write a generic sort solution for events, task, terms and classes. mainly for all entities that have time or date or both

Comment (rhnonose) [reply lib/course_planner/events/events.ex]: We can finish this PR with **2** and discuss **3**

File: lib/course_planner_web/templates/event/index.html.eex [hunk 4]
```diff
@@ -10,74 +10,7 @@
...
-                  <%= link "Delete", to: event_path(@conn, :delete, event), method: :delete, data: [confirm: "Are you sure?"] %>
-                </li>
-              <% end %>
-            </ul>
-          </td>
-        </tr>
-      <% end) %>
-    </tbody>
-  </table>
+<%= render "index_event.html", events: @future_events, section_title: "Future events", conn: @conn %>
```
Comment (ghatighorias) [inline lib/course_planner_web/templates/event/index.html.eex]: what do you think calling it "upcoming events". just a suggestion and i won't block the merge for this indeed

File: lib/course_planner/events/events.ex [hunk 5]
```diff
@@ -17,15 +17,30 @@ defmodule CoursePlanner.Events do
...
+    |> Enum.split_with(&(compare_event_date_time(&1, now)))
+    |> revertse_past_events()
+  end
+
+  def compare_event_date_time(event, now) do
+    {:ok, event_datetime} = NaiveDateTime.new(event.date, event.starting_time)
+    Timex.before?(event_datetime, now)
+  end
+
+  def revertse_past_events({past_events, future_events}) do
```
Comment (ghatighorias) [inline lib/course_planner/events/events.ex]: "revertse" is typo ?
